dspace-api/src/main/java/org/dspace/app/profile/ResearcherProfileServiceImpl.java

```diff
@@ -0,0 +1,317 @@
+
+        if (!item.isArchived() || item.isWithdrawn()) {
+            throw new IllegalArgumentException(
+                "Only archived items can be claimed to create a researcher profile. Item ID: " + item.getID());
+        }
+
+        if (!hasProfileType(item)) {
+            throw new IllegalArgumentException("The provided item has not a profile type. Item ID: " + item.getID());
+        }
+
+        String existingOwner = itemService.getMetadataFirstValue(item, "dspace", "object", "owner", Item.ANY);
+
+        if (StringUtils.isNotBlank(existingOwner)) {
+            throw new IllegalArgumentException("Item with provided uri has already an owner - ID: " + existingOwner);
+        }
+
+        context.turnOffAuthorisationSystem();
+        itemService.addMetadata(context, item, "dspace", "object", "owner", null,
+                                ePerson.getName(), ePerson.getID().toString(), CF_ACCEPTED);
+        context.restoreAuthSystemState();
+
+        return new ResearcherProfile(item);
+    }
+
+    @Override
+    public boolean hasProfileType(Item item) {
+        String profileType = getProfileType();
+        if (StringUtils.isBlank(profileType)) {
+            return false;
+        }
+        return profileType.equals(itemService.getEntityType(item));
+    }
+
+    @Override
+    public String getProfileType() {
+        return configurationService.getProperty("researcher-profile.entity-type", "Person");
+    }
+
+    private Optional<Item> findItemByURI(final Context context, final URI uri) throws SQLException {
+        String path = uri.getPath();
+        UUID uuid = UUIDUtils.fromString(path.substring(path.lastIndexOf("/") + 1));
+        return ofNullable(itemService.find(context, uuid));
+    }
+
+    /**
+     * Search for an profile item owned by an eperson with the given id.
+     */
+    private Item findResearcherProfileItemById(Context context, UUID id) throws SQLException, AuthorizeException {
+
+        String profileType = getProfileType();
+
+        Iterator<Item> items = itemService.findByAuthorityValue(context, "dspace", "object", "owner", id.toString());
+        while (items.hasNext()) {
+            Item item = items.next();
+            String entityType = itemService.getEntityType(item);
+            if (profileType.equals(entityType)) {
+                return item;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns a Profile collection based on a configuration or searching for a
+     * collection of researcher profile type.
+     */
+    @SuppressWarnings("rawtypes")
+    private Optional<Collection> findProfileCollection(Context context) throws SQLException, SearchServiceException {
+        UUID uuid = UUIDUtils.fromString(configurationService.getProperty("researcher-profile.collection.uuid"));
+        if (uuid != null) {
+            return ofNullable(collectionService.find(context, uuid));
```

Comment:
Is it checked anywhere that this collection has an entity type matching the configured value?